### PR TITLE
[improve][pcip] PCIP-1 Add the development process for pulsar-java-contrib project

### PR DESCRIPTION
**Pull request by StevenLuMT**
Add the development process for pulsar-java-contrib project

### CODE REVIEW

File: pcip/TEMPLATE.md
Comment (Denovo1998) [inline pcip/TEMPLATE.md:10]: Images are still needed.

```
IMAGES
If you need the diagrams, please create a folder named pcip-XXX under the pcip/static/img path and put the images in.
```

Comment (StevenLuMT) [reply pcip/TEMPLATE.md]: updated

File: pcip/README.md
Comment (AuroraTwinkle) [inline pcip/README.md:47]: This should be Apache Pulsar Java Contrib project.

File: pcip/TEMPLATE.md
Comment (AuroraTwinkle) [inline pcip/TEMPLATE.md:31]: This is a example from pulsar? Maybe we need one from Pulsar Java Contrib

File: pcip/README.md
Comment (AuroraTwinkle) [inline pcip/README.md:49]: Since we do'nt need DISCUSSION and VOTE as **Development Process** mentioned, is the PCIP workflow here still necessary?

Comment (StevenLuMT) [reply pcip/README.md:49]: Deleted,Thanks

File: pcip/README.md
Comment (StevenLuMT) [reply pcip/README.md:47]: Updated,Thanks

File: pcip/README.md
Comment (liangyepianzhou) [inline pcip/README.md:7]: This repository is for collecting new features and new ideas. Will this involve API changes?

File: pcip/README.md
Comment (liangyepianzhou) [inline pcip/README.md:14]: IMO, the goals can be divided into:
1. As a user manual, add instructions when introducing new features or modifying existing features.
2. Explain the functional design ideas to facilitate review and later maintenance.

File: pcip/README.md
Comment (liangyepianzhou) [inline pcip/README.md:56]: This should also be the Apache Pulsar Java Contrib project. The same goes for the following.

File: pcip/TEMPLATE.md
Comment (liangyepianzhou) [reply pcip/TEMPLATE.md:31]: +1

File: pcip/README.md
Comment (StevenLuMT) [reply pcip/README.md:14]: done

File: pcip/TEMPLATE.md
Comment (StevenLuMT) [reply pcip/TEMPLATE.md]: done

File: pcip/README.md
Comment (StevenLuMT) [reply pcip/README.md]: Updated

File: pcip/README.md
Comment (StevenLuMT) [reply pcip/README.md]: Deleted

File: pcip/README.md
Comment (liangyepianzhou) [inline pcip/README.md:24]:
```suggestion
* Any change to the semantic of existing functionality
```

File: pcip/README.md
Comment (liangyepianzhou) [inline pcip/README.md:31]:
```suggestion
* Documentation changes
```

File: pcip/README.md
Comment (liangyepianzhou) [inline pcip/README.md:32]:
```suggestion
```

File: pcip/README.md
Comment (liangyepianzhou) [inline pcip/README.md:33]:
```suggestion
```

**Comment by AuroraTwinkle** [comment]
LGTM

File: pcip/README.md
Comment (liangyepianzhou) [inline pcip/README.md:61]: I'm thinking we need a PCIP list, which should be maintained in the contributor repository. Then it is best to submit a separate PR to update the PCIP list while the contributor provides PCIP. Add a PCIP whose status is open. The updated PR of this PCIP should be immediately merged by the maintainer without the need for approval by multiple Committers.
Similarly, after PCIP is accepted or rejected, a separate PR should be submitted to update the PCIP list.

Comment (StevenLuMT) [reply pcip/README.md]: You can view all PCIPs (besides the historical ones) as the list of Pull Requests having title starting with `PCIP-`. Here is the [link](https://github.com/apache/pulsar-java-contrib/pulls?q=is%3Apr+%22PCIP-%22+in%3Atitle+sort%3Acreated-desc) for it.

File: pcip/TEMPLATE.md
Comment (liangyepianzhou) [inline pcip/TEMPLATE.md:157]: Delete it.

File: pcip/TEMPLATE.md
Comment (liangyepianzhou) [inline pcip/TEMPLATE.md:160]: Delete it.

File: pcip/TEMPLATE.md
Comment (StevenLuMT) [reply pcip/TEMPLATE.md]: Done

File: pcip/TEMPLATE.md
Comment (StevenLuMT) [reply pcip/TEMPLATE.md]: done

File: pcip/TEMPLATE.md
Comment (liangyepianzhou) [inline pcip/TEMPLATE.md:145]: Delete it

File: pcip/TEMPLATE.md
Comment (liangyepianzhou) [inline pcip/TEMPLATE.md:110]: IMO, maybe it would be better to replace the following parts with quick startup

File: pcip/TEMPLATE.md
Comment (StevenLuMT) [reply pcip/TEMPLATE.md]: done

File: pcip/TEMPLATE.md
Comment (StevenLuMT) [reply pcip/TEMPLATE.md:108]: Done

**Review by liangyepianzhou — APPROVED** [review]
LGTM

**Comment by Denovo1998** [comment]
LGTM

File: pcip/README.md
Comment (eolivelli) [inline pcip/README.md:1]: This is very close to the PIP process.
Do we have a discussion on the dev ML ?

Comment (StevenLuMT) [reply pcip/README.md:1]: mail disscuss link: https://lists.apache.org/thread/z4bx7207cmprvp0vj8mgf1jhs7q2s4vn

**Review by jiazhai — APPROVED** [review]

**Comment by Jason918** [comment]
LGTM

**Review by Jason918 — APPROVED** [review]
LGTM

**Comment by StevenLuMT** [comment]
Mailing List discussion thread: https://lists.apache.org/thread/z4bx7207cmprvp0vj8mgf1jhs7q2s4vn
Mailing List voting thread: https://lists.apache.org/thread/mmd65whp3g9yd6p5n1ytbqws2gt4cmrl

**Comment by StevenLuMT** [comment]
@liangyepianzhou mailing List vote is passed, help me merge this pr

**Pull request merged**
